### PR TITLE
Use globalBanner to render page-specific announcement

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Refactor page-specific announcement
- Use same `globalBanner` as other announcements, it's no longer possible to see 2 different banners at the same time
- Group page-specific banners into `PageBannerTriggers` component
- Move tiny utils with single usage into non-exported implementation details

## Justify why they are needed

This replaces previous PR with the same goal but worse architecture:  https://github.com/HedvigInsurance/racoon/pull/4138

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4164** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4164?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/PageBlock.tsx [hunk 1]
```diff
@@ -22,7 +21,7 @@ export const PageBlock = ({ blok }: PageBlockProps) => {
           <StoryblokComponent blok={nestedBlock} key={nestedBlock._uid} />
         ))}
       </main>
-      <DiscountBannerTrigger />
+      <PageBannerTriggers blok={blok} />
```
Comment (gustaveen) [inline apps/store/src/blocks/PageBlock.tsx:24]: Why don't we pass just the announcement here instead?
```suggestion
      <PageBannerTriggers blok={blok.announcement} />
```

File: apps/store/src/components/Banner/PageBannerTriggers.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  // NOTE: Cleanup has to be separate effect, we need to check current banner to see if `setGlobalBanner` actually succeeded
+  // It's possible that other banner was visible (ex. campaign announcement) and in this case we don't want to remove it
+  const globalBannerId = globalBanner?.id
+  useEffect(() => {
+    return () => {
+      if (globalBannerId != null && globalBannerId === blok?.id) {
+        setGlobalBanner(null)
+      }
+    }
+  }, [blok?.id, globalBannerId, setGlobalBanner])
```
Comment (gustaveen) [inline apps/store/src/components/Banner/PageBannerTriggers.tsx:72]: A bit hard to follow the logic for me, we don't set a banner here if there's already a campaign banner or?

**Review by gustaveen — COMMENTED** [review]
I'm having some problems to follow the logic of the banner setup now, could you maybe walkthrough it?

File: apps/store/src/components/Banner/PageBannerTriggers.tsx (hunk 2 above)
Comment (alebedev) [reply apps/store/src/components/Banner/PageBannerTriggers.tsx:72]: Yes, this is how banner atom works - existing banner is not replaced unless you pass `force: true`. Page announcement banner does not seem important enough for this

**Review by gustaveen — APPROVED** [review]
Great work! 🙌
Things cleared up after walkthrough:
- Had forgotten where we add Announcements in Storyblok (it's done under Page Config)
- Had missed where we consume the global banner atom
